### DISCUSSION TITLE
Hematocrit concern

### DISCUSSION THREAD

**Discussion by christina-rice** [UvaCsl/HemoCell#72] [General discussion]
Does deleting cells that intersect the boundary significantly changes the hematocrit? I am specifically simulating the pipeflow example, and the system seems very dilute compared to healthy blood.

**Comment by gzavo** [comment]
Hi @christina-rice,

It does. It is a limitation of the 'cellPacking' cell initialization method. The cells are initialized in an empty periodic space, and when they are put in a different geometry (e.g. tube), the cells intersecting the walls are removed. Depending on the geometry, this can reduce the resulting hematocrit. In a small diameter tube this can be a significant effect.
Possible ways around it include initializing for larger hematocrit to account for the loss of particles at the boundaries, or creating a cell position file that matches the geometry manually.

**Reply by christina-rice** [reply]
@gzavo I have tried to recreate the tube used in "Red blood cell and platelet diffusivity and margination in the presence of cross-stream gradients in blood flows," but when I create a 50 um diameter tube and load packCell files with hematocrits from .30 - .45, the initialization deletes cells, and I end up with about ~120 cells in a 350 um length tube, which is significantly fewer than expected. 

Is the code from the "Red blood cell and platelet diffusivity and margination in the presence of cross-stream gradients in blood flows" paper available? The pipeflow example seems to be very dilute compared to the publication.

**Comment by mojtabaAmiraslanpour** [comment]
@christina-rice 
I have had the same issue before but by trying 2-3 RBC.pos files (generated by the packCells tool) you will get the hematocrit that you are looking for.

**Reply by christina-rice** [reply]
Thanks for the suggestion. My deletion loss is great enough that switching .pos files doesn't address the issue. Hopefully, I can find the culprit soon and then try swapping various files.

**Reply by mojtabaAmiraslanpour** [reply]
I have tried manually sorting the RBCs in place. It can take some time but excel can come to use to make it easier.

**Comment by mojtabaAmiraslanpour** [comment]
I want to mention a broader issue. This issue does not occur only in the first few iterations but throughout the entire simulation. I am using a curved pipe as the goemetry and red blood cells keep getting deleted at a steady rate. I have plotted the number of cells in my simulation versus time and it doesn't seem to stop.

![image](https://user-images.githubusercontent.com/15649053/197311093-cda3fa77-894d-4a26-843c-208c71dd7f0e.png)

**Reply by gzavo** [reply]
This is a separate problem that should go in a separate discussion.
Anyway, the same list applies as last time: it can be too high shear with too large time-steps, or a problem with the geometry, or a problem with how the pre-inlet is set up. In either case, the first step is to see where the deletions occur, use the detailed info on cell deletion.

**Reply by mojtabaAmiraslanpour** [reply]
So I think I understood the reason for this one. I was using the drivingForce of the straight pipe case for my curved vessel geometry. It would cause RBCs to hit the wall and get deleted. I switched to using the pre-inlet capability.
